doc/buckets.md

```diff
@@ -0,0 +1,145 @@
+# Cloud Bucket configuration
+
+Supported Cloud Providers:
+
+* **gcp**: [Google Storage](#google-storage) _default_
+* **azure-blob**: [Azure Blob Storage](#azure-blob-storage)
+* **s3**: [AWS S3](#aws-s3)
+
+Three different buckets are needed for CARTO Self Hosted: two for imports and one to store maps thumbnails.
+
+Map thumbnails can be configured in two ways. Public (thumbnails are public) or private and signed URLs are generated for them. In order to control it you need to change the variable: `WORKSPACE_THUMBNAILS_PUBLIC='true'`.
+
+## Google Storage
+
+You need 3 buckets and a Service Account that can read/write to them.
+
+### Thumbnails
+
+```bash
+WORKSPACE_THUMBNAILS_PROVIDER='gcp'
+WORKSPACE_THUMBNAILS_BUCKET='bucket-name1'
+WORKSPACE_THUMBNAILS_KEYFILENAME='route_to_json'
+WORKSPACE_THUMBNAILS_PROJECTID='myproject'
+```
+
+To be able to sign URLs, the Service Account used needs to have the permission `iam.serviceAccounts.signBlob` on the bucket.
+
+### Imports
+
+The Service Account used needs to have the permission `iam.serviceAccounts.signBlob` on this bucket.
+
+```bash
+WORKSPACE_IMPORTS_PROVIDER='gcp'
+WORKSPACE_IMPORTS_BUCKET='bucket-name2'
+WORKSPACE_IMPORTS_KEYFILENAME='route_to_json'
+WORKSPACE_IMPORTS_PROJECTID='myproject'
+```
+
+```bash
+IMPORT_PROVIDER='gcp'
+IMPORT_BUCKET='bucket-name3'
+IMPORT_KEYFILENAME='route_to_json'
+IMPORT_PROJECTID='myproject'
+```
+
+### Notes
+
+if `_KEYFILENAME` is not defined  env `GOOGLE_APPLICATION_CREDENTIALS` is used as default value
+
+if `_PROJECTID` is not defined  env `GCLOUD_PROJECT` is used as default value
+
```

Comment:
## to add
if neither `_KEYFILENAME` nor `GOOGLE_APPLICATION_CREDENTIALS` are provided (meaning the container is inheriting default credentials from an instance), the service account needs one additional permission at the project level: ` iam.serviceAccounts.signBlob` (contained by the role `roles/iam.serviceAccountTokenCreator`)